spec.html

```diff
@@ -14875,16 +14875,19 @@ <h1>
         <emu-alg>
           1. Let _O_ be _taRecord_.[[Object]].
           1. Let _bufferByteLength_ be _taRecord_.[[CachedBufferByteLength]].
-          1. Assert: IsDetachedBuffer(_O_.[[ViewedArrayBuffer]]) is *true* if and only if _bufferByteLength_ is ~detached~.
-          1. If _bufferByteLength_ is ~detached~, return *true*.
+          1. If IsDetachedBuffer(_O_.[[ViewedArrayBuffer]]) is *true*, then
+            1. Assert: _bufferByteLength_ is ~detached~.
+            1. Return *true*.
+          1. Assert: _bufferByteLength_ is a non-negative integer.
           1. Let _byteOffsetStart_ be _O_.[[ByteOffset]].
           1. If _O_.[[ArrayLength]] is ~auto~, then
             1. Let _byteOffsetEnd_ be _bufferByteLength_.
           1. Else,
             1. Let _elementSize_ be TypedArrayElementSize(_O_).
-            1. Let _byteOffsetEnd_ be _byteOffsetStart_ + _O_.[[ArrayLength]] × _elementSize_.
+            1. Let _arrayByteLength_ be _O_.[[ArrayLength]] × _elementSize_.
+            1. Let _byteOffsetEnd_ be _byteOffsetStart_ + _arrayByteLength_.
```

Comment:
Why split this up into two aliases? I like the original arithmetic expression better.

Reply:
Better symmetry with the analogous IsViewOutOfBounds.

Reply:
Fair enough.